File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    static Stack<Character> stack = new Stack<>();
    boolean solution(String s) {
        char[] arr = s.toCharArray();
        for(char ch : arr){
            if(ch=='('){
                stack.add('(');
            }else{
                if(stack.isEmpty()){
                    return false;
                }else{
                    stack.pop();
                }
            }
        }
        if(stack.isEmpty()){
            return true;
        }else{
            return false;
        }

    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<char> stack ;
    for(int i=0; i<s.size(); i++){
        if(i==0 && s[i]==')'){
            return false;
        }
        if(s[i]==')'){
            if(!stack.empty()&&stack.top()=='('){
                stack.pop();
            }else{
                return false;
            }
        }else{
            stack.push(s[i]);
        }
    }
    
    if(stack.empty())
        return true;
    return false;

    
    return answer;
}